.gitlab-ci.yml

```diff
@@ -30,11 +30,30 @@ build-runner-image:
     - docker buildx build --no-cache --pull --push --label target=build --tag ${CI_REGISTRY_IMAGE_TEST}:${CI_COMMIT_SHORT_SHA} --tag ${CI_REGISTRY_IMAGE_TEST}:${CI_COMMIT_SHA:0:12} --secret id=github_token,env=GITHUB_TOKEN .
   retry: 2
 
+build-pulumi-go-main:
+  stage: build
+  image: 486234852809.dkr.ecr.us-east-1.amazonaws.com/ci/datadog-agent-buildimages/deb_x64:v38375519-53773733
+  tags: ["arch:amd64"]
+  rules:
+    - when: on_success
+  script:
+    - go build -o dist/main -gcflags="all=-c=6" main.go 
+  artifacts:
+    paths:
+      - dist/main
+    expire_in: "1 day"
```

Comment:
Do you think we should add a warning if we try to rerun the integration test and the prebuilt artifact is no longer available?

Reply:
Yes, good idea. I can add a check at job time to ensure that the `dist/main` is still there, and recommend re-running the prev job on failure

Reply:
Added